src/liballoc/vec.rs

```diff
@@ -491,7 +491,9 @@ impl<T> Vec<T> {
     ///
     /// # Panics
     ///
-    /// Panics if the new capacity overflows `usize`.
+    /// * Panics if the requested capacity exceeds `usize::MAX` bytes.
+    /// * Panics on 32-bit platforms if the requested capacity exceeds
```

Comment:
This is a bit misleading -- and the (documentation) commentary in RawVec could be updated as well. As far as I can tell, even though the code special cases platforms that are <64-bit, that's just done as a performance optimization: we expect that you *cannot* allocate isize::MAX sized objects on a 64-bit (or larger, I guess) platform, because they would simply be "too big." That seems like a reasonable assumption to make.

However, calling out 32-bit here seems confusing: I would prefer to instead simply say that the new capacity cannot exceed `isize::MAX` bytes, which I think must always be correct.

Reply:
Yes, I feel like this is misleading as well but the old documentation uses that. I guess I need to go back and change the old RawVec documentation.